### ISSUE TITLE
No way to use a multi-word git tag message?

### ISSUE THREAD

**Issue by ruohola** [EndBug/add-and-commit#166]
There doesn't seem to be any way to pass a `--message` that has multiple word to the Git tag command. For example this attempt:

      - name: "Commit changes"
        uses: EndBug/add-and-commit@v7
        with:
          add: 'CHANGELOG.md'
          message: 'Release 1.0.0'
          tag: "v1.0.0 --annotate --message 'Release 1.0.0'"

Fails with:

```
2021-04-13T10:51:08.1831581Z > Tagging commit...
2021-04-13T10:51:08.1889446Z ##[error]Error: fatal: Failed to resolve '1.0.0'' as a valid ref.

2021-04-13T10:51:08.1904561Z ##[error]Error: fatal: Failed to resolve '1.0.0'' as a valid ref.

2021-04-13T10:51:08.1908826Z > Pushing commit to repo...
```

and this:

     - name: "Commit changes"
        uses: EndBug/add-and-commit@v7
        with:
          add: 'CHANGELOG.md'
          message: 'Release 1.0.0'
          tag: 'v1.0.0 --annotate --message "Release 1.0.0"'

results in the tag message being literally `"Release 1.0.0"`.

I tried all kinds of YAML quoting, but the problem is really this function:

https://github.com/EndBug/add-and-commit/blob/f4b07e6873721ec26cd5d31d795fe378f2855004/src/util.ts#L41-L43

There is no way to pass arguments to that so that it would keep the `Release 1.0.0` as one string with no extra quotes around it:

https://www.typescriptlang.org/play?ts=4.2.3#code/GYVwdgxgLglg9mABAWwIZQgCwOIygQQCcBzAZwApSpCYxiAuRKmugSkQG8AoRRQgUyghCSZrWIA6NBkzkA9OQD89ANoA9ADqkARAF0A1AB9t6vQCptrfXOLtDhxCt1cAvly4QEpOABt+EnzhicmksXAISCm0ANwBGCQAGRMQAWhTUMDA4KHR+VJTkflJSVGI8gHIAJX4-VFI8+KSE8stWAG4PL19-QODQnDwiMnJyuMTktIysnKg8tMLi0rztatr6xEbE7XLWds6wbz8AoJD0MMHI8hjNhPyp7Nz8hZKyxA0Vmv46hvGE99aOp4Dt1jn0zgMIsNRjc7pkHrMnkUXnkNFVPt8Nr9UTt2kA

**Comment by ruohola** [comment]
I solved this for now by writing the tag message to a file in a previous step and then using `tag: v1.0.0 --annotate --file=message.txt` in the `add-and-commit` step.

**Comment by EndBug** [comment]
Happy to see that you found a workaround :)
Anyway, I'll try to see if I can fix it as soon as I can

**Comment by EndBug** [comment]
Hi, I think I've fixed the issue. I've edited the function to remove double quotes:
https://github.com/EndBug/add-and-commit/blob/2d95cfd00a2740421b062b5c8bfdecbc93da17c2/src/util.ts#L41-L46

Sorry it took so long but I had to troubleshoot an issue with the compiler dependency. I'd like you to test this fix by temporarily using the `git-args` branch:

```yaml
- uses: EndBug/add-and-commit@git-args
  ...
```

Let me know if that solves the issue!

**Comment by ruohola** [comment]
> Hi, I think I've fixed the issue. I've edited the function to remove double quotes:
> 
> https://github.com/EndBug/add-and-commit/blob/2d95cfd00a2740421b062b5c8bfdecbc93da17c2/src/util.ts#L41-L46
> 
> Sorry it took so long but I had to troubleshoot an issue with the compiler dependency. I'd like you to test this fix by temporarily using the `git-args` branch:
> 
> ```yaml
> - uses: EndBug/add-and-commit@git-args
>   ...
> ```
> 
> Let me know if that solves the issue!

😅 actually it turned out that for my current use case the file method worked better, since the message will be quite long. Great if you got this fixed though since the current behavior is of course problematic. 💪

**Comment by ruohola** [comment]
Not sure if removing double quotes is a good solution, since then one couldn't use `--message 'This is "quoted" message'` at all. Might not be trivial to solve that one though.

**Comment by EndBug** [comment]
You're right. Thanks to a guy on Stack Overflow I've gotten to this solution: 

https://github.com/EndBug/add-and-commit/blob/23d2118353b628d19183fb63a819316b7ec06aaf/src/util.ts#L33-L60

[Here](https://replit.com/@FedericoGrandi/matchGitArgs#index.js)'s an interactive version. This should correctly parse every argument. 

Anyway, I agree that if you're going to use a very long commit message, using a file is the best solution.

**Comment by ruohola** [comment]
> [Here](https://replit.com/@FedericoGrandi/matchGitArgs#index.js)'s an interactive version. This should correctly parse every argument.

Looks solid 👍

**Comment by EndBug** [comment]
@all-contributors please add @ruohola for their bug report

**Comment by EndBug** [comment]
Fix has been published in v7.1.2 (also v7 and latest). Thanks again for the report!
